packages/blobdex/src/index.js

```diff
@@ -0,0 +1,81 @@
+import { base58btc } from 'multiformats/bases/base58'
+export * from './types.js'
+
+export class Blobdex {
+  #items
+
+  constructor() {
+    this.#items = new Map()
+  }
+
+  /**
+   * len numbers the count of items in this Blobdex.
+   *
+   * @returns {number}
+   */
+  len() {
+    return this.#items.size
+  }
+
+  /**
+   * add adds a new item to this Blobdex multihash digest bytes key.
+   *
+   * @param {import('multiformats').MultihashDigest} mh
+   * @param {number} offset
+   * @param {number} length
+   */
+  add(mh, offset, length) {
+    const b58str = base58btc.encode(mh.bytes)
+    this.addStr(b58str, offset, length)
+  }
+
+  /**
+   * addStr adds a new item to this Blobdex with base-58 key.
+   *
+   * @param {string} b58Str
+   * @param {number} offset
+   * @param {number} length
+   */
+  addStr(b58Str, offset, length) {
```

Comment:
When might you need this? This seems a little error prone and exposes the internals of the encoding - we're also not doing any validation so could cause errors on read. Personally I'd not to expose this publically.